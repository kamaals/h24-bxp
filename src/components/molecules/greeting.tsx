import React from "react";
import { headers } from "next/headers";
import { auth } from "@/lib/auth";
import Link from "next/link";
import Logout from "@/components/molecules/auth/logout";
import { LogInIcon } from "lucide-react";

export async function GreetingMenu() {
  const session = await auth.api.getSession({
    headers: await headers(),
  });
  if (!session) {
    return (
      <ul className="flex gap-4 items-center">
        <li>
          <Link href={"/register"}>Register</Link>
        </li>
        <li className={"flex items-center gap-2"}>
          <LogInIcon size={16} />
          <Link href={"/login"}>Login</Link>
        </li>
      </ul>
    );
  }
  return (
    <ul className="flex gap-4 items-center">
      <li>
        <Link href={"/register"}>Dashboard</Link>
      </li>
      <li>
        <Logout />
      </li>
    </ul>
  );
}

async function Greeting() {
  const session = await auth.api.getSession({
    headers: await headers(),
  });
  if (!session) {
    return (
      <h1 className="text-4xl">
        Hej! Please <Link href={"/login"}>Login</Link> to continue
      </h1>
    );
  }
  return (
    <h1 className="text-4xl">
      Hej! welcome back, please continue to{" "}
      <Link href={"/dashboard"}>Dashboard</Link>
    </h1>
  );
}

export default Greeting;
