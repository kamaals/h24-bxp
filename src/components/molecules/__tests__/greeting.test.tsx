import { describe, expect, it, vi } from "vitest";
import { render } from "@testing-library/react";
import { UserGreet, AnonGreet, NavLoggedIn, NavLoggedOut } from "../greeting";
import React from "react";

vi.mock("next/navigation", () => ({
  useRouter: () => ({
    push: vi.fn(),
    refresh: vi.fn(),
  }),
}));

describe(UserGreet.name, () => {
  it("should match the snapshot", () => {
    const result = render(<UserGreet />);
    expect(result).toMatchSnapshot();
  });
});

describe(AnonGreet.name, () => {
  it("should match the snapshot", () => {
    const result = render(<AnonGreet />);
    expect(result).toMatchSnapshot();
  });
});

describe(NavLoggedIn.name, () => {
  it("should match the snapshot", () => {
    const result = render(<NavLoggedIn />);
    expect(result).toMatchSnapshot();
  });
});

describe(NavLoggedOut.name, () => {
  it("should match the snapshot", () => {
    const result = render(<NavLoggedOut />);
    expect(result).toMatchSnapshot();
  });
});
