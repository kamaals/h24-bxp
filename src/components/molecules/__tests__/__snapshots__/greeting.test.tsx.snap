// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AnonGreet > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-4xl"
      >
        Hej! welcome back, please continue to
         
        <a
          href="/dashboard"
        >
          Dashboard
        </a>
      </h1>
    </div>
    <div>
      <h1
        class="text-4xl"
      >
        Hej! welcome back, please continue to
         
        <a
          href="/dashboard"
        >
          Dashboard
        </a>
      </h1>
    </div>
  </body>,
  "container": <div>
    <h1
      class="text-4xl"
    >
      Hej! welcome back, please continue to
       
      <a
        href="/dashboard"
      >
        Dashboard
      </a>
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavLoggedIn > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-4xl"
      >
        Hej! welcome back, please continue to
         
        <a
          href="/dashboard"
        >
          Dashboard
        </a>
      </h1>
    </div>
    <div>
      <h1
        class="text-4xl"
      >
        Hej! welcome back, please continue to
         
        <a
          href="/dashboard"
        >
          Dashboard
        </a>
      </h1>
    </div>
    <div>
      <ul
        class="flex gap-4 items-center"
      >
        <li>
          <a
            href="/dashboard"
          >
            Dashboard
          </a>
        </li>
        <li>
          <div>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 h-9 px-4 py-2 has-[>svg]:px-3"
              data-slot="button"
              data-testid="logout-btn"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-log-out"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                />
                <polyline
                  points="16 17 21 12 16 7"
                />
                <line
                  x1="21"
                  x2="9"
                  y1="12"
                  y2="12"
                />
              </svg>
               Logout
            </button>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex gap-4 items-center"
    >
      <li>
        <a
          href="/dashboard"
        >
          Dashboard
        </a>
      </li>
      <li>
        <div>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 h-9 px-4 py-2 has-[>svg]:px-3"
            data-slot="button"
            data-testid="logout-btn"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-log-out"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
              />
              <polyline
                points="16 17 21 12 16 7"
              />
              <line
                x1="21"
                x2="9"
                y1="12"
                y2="12"
              />
            </svg>
             Logout
          </button>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavLoggedOut > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-4xl"
      >
        Hej! welcome back, please continue to
         
        <a
          href="/dashboard"
        >
          Dashboard
        </a>
      </h1>
    </div>
    <div>
      <h1
        class="text-4xl"
      >
        Hej! welcome back, please continue to
         
        <a
          href="/dashboard"
        >
          Dashboard
        </a>
      </h1>
    </div>
    <div>
      <ul
        class="flex gap-4 items-center"
      >
        <li>
          <a
            href="/dashboard"
          >
            Dashboard
          </a>
        </li>
        <li>
          <div>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 h-9 px-4 py-2 has-[>svg]:px-3"
              data-slot="button"
              data-testid="logout-btn"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-log-out"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                />
                <polyline
                  points="16 17 21 12 16 7"
                />
                <line
                  x1="21"
                  x2="9"
                  y1="12"
                  y2="12"
                />
              </svg>
               Logout
            </button>
          </div>
        </li>
      </ul>
    </div>
    <div>
      <ul
        class="flex gap-4 items-center"
      >
        <li>
          <a
            href="/register"
          >
            Register
          </a>
        </li>
        <li
          class="flex items-center gap-2"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-log-in"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4"
            />
            <polyline
              points="10 17 15 12 10 7"
            />
            <line
              x1="15"
              x2="3"
              y1="12"
              y2="12"
            />
          </svg>
          <a
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex gap-4 items-center"
    >
      <li>
        <a
          href="/register"
        >
          Register
        </a>
      </li>
      <li
        class="flex items-center gap-2"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-log-in"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4"
          />
          <polyline
            points="10 17 15 12 10 7"
          />
          <line
            x1="15"
            x2="3"
            y1="12"
            y2="12"
          />
        </svg>
        <a
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UserGreet > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-4xl"
      >
        Hej! welcome back, please continue to
         
        <a
          href="/dashboard"
        >
          Dashboard
        </a>
      </h1>
    </div>
  </body>,
  "container": <div>
    <h1
      class="text-4xl"
    >
      Hej! welcome back, please continue to
       
      <a
        href="/dashboard"
      >
        Dashboard
      </a>
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
